paymentsheet/src/main/java/com/stripe/android/link/account/DefaultLinkAccountManager.kt

```diff
@@ -547,25 +549,54 @@ internal class DefaultLinkAccountManager @Inject constructor(
         linkAccount: LinkAccount?,
         canLookupCustomerEmail: Boolean
     ): AccountStatus {
-        // If we already have an account, return its status.
-        if (linkAccount != null) {
-            return linkAccount.accountStatus
-        }
-        // Look up the customer email if possible.
-        return config.customerInfo.email?.takeIf { canLookupCustomerEmail }
-            ?.let { customerEmail ->
-                lookupConsumer(
-                    email = customerEmail,
-                    linkAuthIntentId = null,
-                    startSession = true,
-                    customerId = config.customerIdForEceDefaultValues
-                )
-                    .map { it?.accountStatus }
-                    .getOrElse { AccountStatus.Error }
+        val result =
+            when (val linkLaunchMode = this.linkLaunchMode) {
+                null,
+                is LinkLaunchMode.Authentication,
+                is LinkLaunchMode.Confirmation,
+                LinkLaunchMode.Full,
+                is LinkLaunchMode.PaymentMethodSelection -> {
+                    linkAccount
+                        // If we already have an account, return it.
+                        ?.let { Result.success(it) }
+                        ?: run {
+                            val email = config.customerInfo.email?.takeIf { canLookupCustomerEmail }
+                            email?.let { lookupConsumerByEmail(it) }
+                        }
+                }
+                is LinkLaunchMode.Authorization -> {
+                    val linkAuthIntentId = linkLaunchMode.linkAuthIntentId
+                    linkAccount
+                        // If we already have an account for the LAI, return it.
+                        ?.takeIf { it.linkAuthIntentInfo?.linkAuthIntentId == linkAuthIntentId }
+                        ?.let { Result.success(it) }
+                        ?: lookupConsumerByAuthIntent(linkAuthIntentId)
+                }
```

Comment:
so `Authorization` will always be auth with LAI in the near term right? we can probably have an `AuthorizationMode` in the future (`lai`, `email`)